pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -201,6 +202,9 @@ def __repr__(self) -> str:
             state += f", cumulated_batch_size={self.cumulated_batch_size}"
         return f"{self.__class__.__name__}({state})"
 
+    def __getstate__(self) -> dict:
+        return {**super().__getstate__(), '_class': self.__class__.__name__}
```

Comment:
Should we care about pickling the `_Metadata` and `_Sync` dataclasses?

Reply:
I think so. And we should drop the sync.fn function too.

Reply:
Done